### DISCUSSION TITLE
Is anyway to call API inside Orchard Core CMS ?

### DISCUSSION THREAD

**Discussion by hoanghai1212** [OrchardCMS/OrchardCore#7922] [Q&A] [answered]
Hi Everyone, I want to design a website (called A), there will be static components on this page (I intend to use HTML returned by GraphQL from OrchardCMS) and some of them will fetch data from another web API. Currently I am considering 2 options:
1. Using Full CMS from Orchard core CMS: but I don't know how to fetch data from another web API from OrchardCMS. Is there any way to do this?
2. Using Orchard core CMS as a headless CMS: I fetch data from CMS through GraphQL API then mix with data from another web API as mentioned.
=> Which one is the best option, is there any other choice ? Thank you

**Comment by jptissot** [comment, marked as answer]
I was able to do this using a workflow in the past. But it's hard to know without your exact requirement. Look at the Http Request workflow task and see if it can work for your use case.

**Reply by hoanghai1212** [reply]
I want to build a page that will contain main 2 thing:
1. Some static post that I won't change the post usually, but have some page will contain daily post.
2. Search feature that get data from another API.
For now I use CMS for daily post and considering between take static post from Orchard Core CMS through htmlBody( I just use html code response from GraphQL) and hardcode the static post in Frontend (Reactjs)
